packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -4799,10 +4800,26 @@ class RichText extends LeafRenderObjectWidget {
        assert(overflow != null),
        assert(textScaleFactor != null),
        assert(maxLines == null || maxLines > 0),
-       super(key: key);
+       super(key: key, children: _extractChildren(text));
+
+  // Traverses the InlineSpan tree and depth-first collects the list of
+  // child widgets that are created in WidgetSpans.
+  static List<Widget> _extractChildren(InlineSpan span) {
+    List<Widget> result = [];
+    span.visitChildren((InlineSpan span) {
+      if (span is! WidgetSpan)
+        return true;
+      WidgetSpan widgetSpan = span as WidgetSpan;
```

Comment:
this line should not be necessary. (if it is, try using `if ...is... { ... }` rather than `if ...is!... return; ...`)